tasks/run-matlab-tests/v1/test/scriptgen.test.ts

```diff
@@ -0,0 +1,90 @@
+            assert(actual.includes("'SourceFolder',''"));
+            assert(actual.includes("'PDFTestReport',''"));
+            assert(actual.includes("'SimulinkTestResults',''"));
+            assert(actual.includes("'CoberturaModelCoverage',''"));
+            assert(actual.includes("'SelectByTag',''"));
+            assert(actual.includes("'SelectByFolder',''"));
+            assert(actual.includes("'Strict',false"));
+            assert(actual.includes("'UseParallel',false"));
+            assert(actual.includes("'OutputDetail',''"));
+            assert(actual.includes("'LoggingLevel',''"));
+
+            const expected = `genscript('Test', 'JUnitTestResults','', 'CoberturaCodeCoverage','',
+                'SourceFolder','', 'PDFTestReport','', 'SimulinkTestResults','',
+                'CoberturaModelCoverage','', 'SelectByTag','', 'SelectByFolder','',
+                'Strict',false, 'UseParallel',false, 'OutputDetail','', 'LoggingLevel','')`
+                .replace(/\s+/g, "");
+            assert(actual.replace(/\s+/g, "").includes(expected));
+        });
+
+        it("contains genscript invocation with all options specified", () => {
+            const options: scriptgen.IRunTestsOptions = {
+                JUnitTestResults: "test-results/results.xml",
+                CoberturaCodeCoverage: "code-coverage/coverage.xml",
+                SourceFolder: "source",
+                PDFTestReport: "test-results/pdf-results.pdf",
+                SimulinkTestResults: "test-results/simulinkTest.mldatx",
+                CoberturaModelCoverage: "test-results/modelcoverage.xml",
+                SelectByTag: "FeatureA",
+                SelectByFolder: "test/tools;test/toolbox",
+                Strict: true,
+                UseParallel: true,
+                OutputDetail: "Detailed",
+                LoggingLevel: "Detailed",
+            };
+
+            const actual = scriptgen.generateCommand(options);
+
+            assert(actual.includes("genscript('Test'"));
+            assert(actual.includes("'JUnitTestResults','test-results/results.xml'"));
+            assert(actual.includes("'CoberturaCodeCoverage','code-coverage/coverage.xml'"));
+            assert(actual.includes("'SourceFolder','source'"));
+            assert(actual.includes("'PDFTestReport','test-results/pdf-results.pdf'"));
+            assert(actual.includes("'SimulinkTestResults','test-results/simulinkTest.mldatx'"));
+            assert(actual.includes("'CoberturaModelCoverage','test-results/modelcoverage.xml'"));
+            assert(actual.includes("'SelectByTag','FeatureA'"));
+            assert(actual.includes("'SelectByFolder','test/tools;test/toolbox'"));
+            assert(actual.includes("'Strict',true"));
+            assert(actual.includes("'UseParallel',true"));
+            assert(actual.includes("'OutputDetail','Detailed'"));
+            assert(actual.includes("'LoggingLevel','Detailed'"));
```

Comment:
Unnecessary with the check  below?

Reply:
Yeah maybe. I think I prefer this way so we know which one is causing the problem. Otherwise it's a lot of text to figure out where the failure occurs. I think I prefer leaving these or both (I don't think both hurts).

Reply:
Gotcha, having both works for me!